### ISSUE TITLE
Make Slick Flow variable in return type

### ISSUE THREAD

**Issue by diversit** [akka/alpakka#726]
Currently the Slick.flow signature fixes the return type to Int.
`Slick.flow(..): Flow[T, Int, NotUsed]`

However sometimes it might be useful to be able to return a different type.
Case: when consuming from a Kafka topic and you want to commit the message once the message has been persisted.
In the current situation this is not possible since you lose the CommittableMessage or Committable after using the Slick Flow.
It might be possible to split the flow and rejoin again after persisting via Slick, but this greatly complicates a very simple flow.

Proposed solution:
Make the Slick.flow return type dynamic. This would allow to do something like:
```
Slick.flow { committableMessage =>
  val myRow = msgToRow(committableMessage)
  (Tables.MyTable += row).map(insertCount => (committableMessage, insertCount))
}
```
The committable message is now retained and can be committed in a next flow step.

Alternative:
Add a additional 'flow' function which does not fix the return type. But what would then be a good name for this function?

If this change is/gets approved, I will make a pull request.

**Comment by ennru** [comment]
Good idea.
We see this need in other connectors, as well.
The Elasticsearch flow got support in the latest release and mimics the Reactive Kafka naming of a `passThrough`.

**Comment by diversit** [comment]
Created a proposal for this feature in pull request #743

**Comment by ennru** [comment]
Fixed by #743
